src/drivers/ll40ls/LidarLitePWM.cpp

```diff
@@ -196,6 +200,27 @@ int LidarLitePWM::measure()
 		return reset_sensor();
 	}
 
+	uint32_t pulse_interval = _range.timestamp - _lastTimeStamp;
+	_lastTimeStamp = _range.timestamp;
+
+	// Normal reporting interval for LidarLite in PWM mode is 50msec
+	if (pulse_interval > 60 * 1000UL) { // TODO : Should we increase this to *not* miss longer range meas?
+		// missed pulse, frequently associated with invalid pulsewidth measurements
+		_pulseCount = 0;
+
+	} else {
```

Comment:
This is from @kd0aij actually. Not 100% LPE-specific, but fixes the weird LidarLite ground spikes (which cause a false lidar init and a bad altitude estimate, Z velocity fails, etc.) Since LPE relies a lot on the Lidar for precise altitude estimation, this is necessary when taking off or landing to prevent crazy stuff. I crashed twice before figuring out the issue.

But lets look into fault detection.